### DISCUSSION TITLE
0.97.2.4 for MC 1.14.* & 1.15.* & 1.16.* & 1.17.*

### DISCUSSION THREAD

**Discussion by LlmDl** [TownyAdvanced/Towny#5334] [Pre-Release Discussions]
- Fix Global messages stripping their colour.
  - Remove old Outpost checking code and references to Articdive which were not spelled correct, courtesy of Articdive with PR #5324.
  - Add a size limit for the town_block_ratio option, courtesy of Warriorrrr with PR #5326.
  - New Config Option: town.town_block_limit
    - Default: 0
    - The maximimum amount of townblocks a town can have, if town_block_ratio is 0 the max size will be decided by the town_levels. 
    - Set to 0 to have no size limit.
  - Fix automatic renaming of invalid and duplicate town and nation names failing some of the time.
  - Fix EventWar not handling the OutlawTeleportEvent.
  - Fix resident/town counts being shown on the status screens when "and more..." was being used on long lists.
    - Closes #5331. 
  - Lists of TownyObjects (resident lists, town lists, nation lists) will no longer use their longer formatted names if the list is over 19.
    - Closes #5332. 
  - Fix Status Screen components' hovering bleeding over into blank spaces further along the line.
    - Closes #5332.
    
----

<details><summary>Cumulative changes since 0.97.2.0</summary>
<details><summary>0.97.2.3</summary>

  - Change PlayerCacheUtil to once again check for a fakeplayer resident.
  - Fix PlayerCacheUtil preventing plot permissions from working properly. 
  - Fix missing command feedback on many /tw toggle commands.
</details>
<details><summary>0.97.2.2</summary>

  - Fix Citizens NPCs causing error on PlayerTeleportEvent, courtesy of ArticDive with PR #5319.
</details>
<details><summary>0.97.2.1</summary>

  - Tidy up PlayerCacheUtil.
  - Fix missing splitter between Nation bank and upkeep on the nation status screen.
  - New Placholder: %townyadvanced_town_board%
    - Displays a player's town's board message.
  - New Placeholder: %townyadvanced_nation_board%
    - Displays a player's nation's board message.
  - New Placeholder: %townyadvanced_player_location_town_board%
    - Displays the board message of the town where the player is stood.
  - New Placeholder: %townyadvanced_player_location_nation_board%
    - Displays the board message of the nation where the player is stood.
  - Add spawn protection for players who've just died.
  - New Config Node: global_town_settings.respawn_protection
    - Default; 10s
    - When greater than 0s, the amount of time a player who has respawned is considered invulnerable.
    - Invulnerable players who attacker other players will lose their invulnerability.
    - Invulnerable players who teleport after respawn will also lose their invulnerability.
  - Bump language files to 0.125.
  - Prevent spam if a player's locale is invalid, courtesy of Warriorrrr with PR #5314.
  - Improve startup logging/initialization, courtesy of ArticDive & LlmDl with PR #5299.
</details>
</details>

----

<table align=center>
<tr><th colspan=4><h3>Towny Sponsors<br><h4>I want to give a big thank you to all of my sponsors:<br>These are the people that help to make Towny's support and development as active as it is,<br>and who supported me during this pre-release of Towny. </th></tr>
<tr><td>MrLogano</td><td>Momshroom</td><td>Cobrex1</td><td> KarlOfDuty </td></tr>
<tr><td>scatterspell</td><td>TheFlagCourier</td><td>gerardvanschip</td><td>DrprofLuigi</td></tr>
<tr><td>Tobza</td><td>SmallSansSerif</td><td>TheNylox</td><td>diontimmer</td></tr>
<tr><td>trevor1097</td><td>benevolent2012</td><td>bretomart</td><td>HunSolos</td></tr>
<tr><td>0xBit-MC</td><td>specs1396</td><td>treeDerevo</td><td>Scottlander100</td></tr>
<tr><td>ItsSynchro</td><td>MidnightSugar</td><td>PigioGigio</td><td>PestoNotPasta</td></tr>
<tr><td>catsaretheend</td><td>cigaming</td><td>KeNickGong</td><td>honnisha</td></tr>
<tr><td>MilesBHuff</td><td>pebblehost</td><td>Akiranya</td><td>BamBamTheEggman</td></tr>
<tr><td>TakeoRenka</td><td>LordHenryVonHenry</td><td>nUSAHudson</td><td>GORKEMCNC</td></tr>
<tr><td>SolarCiv</td><td colspan=3>and 7 private sponsors.</td></tr>
<tr><td colspan=4 align=center><h3><a href=https://github.com/sponsors/LlmDl>If you want to support the developer consider becoming a sponsor.</a><br><h6>(It's just like Patreon but 100% goes to the developer.)<br><h3>Plus there are <a href="https://github.com/TownyAdvanced/Towny/wiki/Sponsor-Perks"><b>perks!</b></a></td></tr>
</table>

<table align=center>
<th colspan=3> Important Links </th>
<tr align=center><td><a title="How Towny Works" href="https://github.com/TownyAdvanced/Towny/wiki/How-Towny-Works">How Towny 
Works</a></td><td><a title="Install Guide" href="https://github.com/TownyAdvanced/Towny/wiki/Installation">Towny Install Guide</a></td><td><a title="Updating Towny" href="https://github.com/TownyAdvanced/Towny/wiki/Updating-Towny">Towny Update Guide</a></td>
</tr>
<tr align=center>
<td><a title="Towny Changelog" href="https://raw.githubusercontent.com/TownyAdvanced/Towny/master/resources/ChangeLog.txt" target="_blank" rel="noopener">Complete Changelog</a> </td><td><a title="Default Config Files" href="https://github.com/TownyAdvanced/Towny/wiki/Config-Files">Default Config Files</a></td><td><a title="Commands/Permissions/Placeholders" href="https://github.com/TownyAdvanced/Towny/wiki/Reference">Commands/Permissions<br>Placeholders</a></td>
</tr>
</table>

### ⏬ Download available as a .jar file in the Assets section below:

<hr /><em>This discussion was created from the release <a href='https://github.com/TownyAdvanced/Towny/releases/tag/0.97.2.4'>0.97.2.4 for MC 1.14.* & 1.15.* & 1.16.* & 1.17.*</a>.</em>

**Comment by SkapMC** [comment]
pog
